contracts/EpochTokenLocker.sol

```diff
@@ -123,6 +123,14 @@ contract EpochTokenLocker {
     /**
      * internal functions
      */
+    function addBalanceAndPostponeWithdraw(address user, address token, uint amount) internal {
+        updateStateIndex();
+        if (balanceStates[user][token].pendingWithdraws.stateIndex < currentStateIndex) {
```

Comment:
Does this couple of lines say "if state index was incremented, also increment the withdraw claim index?". If so, could we achieve the same result by using   `+= 1` as well?

Reply:
nit: Given that stateIndex can never be > currentStateIndex (it can only be = or <), should we just always set it to =?

Reply:
>Does this couple of lines say "if state index was incremented, also increment the withdraw claim index?". If so, could we achieve the same result by using += 1 as well?

no, we always want to push it to the current state index. Even the pendingWithdraws.stateIndex is much smaller

>nit: Given that stateIndex can never be > currentStateIndex (it can only be = or <), should we just always set it to =?

The way I coded it, it is a little bit more gas efficient.

Reply:
> The way I coded it, it is a little bit more gas efficient.

Did you benchmark this? I wrote the following smart contract:

```solidity
contract OverrideTest {
    uint bar;
    function set(uint b) public {
        bar = b;
    }
    function set_cheap(uint b) public {
        if (bar != b) {
            bar = b;   
        }
    }
}
```

And ran

```
set(1) // execution cost: 20220 gas (since first write)
set(1) // execution cost: 420 gas
set_cheap(1) // execution cost: 451 gas
```
Looks like set_cheap is actually slightly more expensive. I believe the optimizer is doing a good job at only writing when necessary.

Reply:
Oh, interesting. Thanks for the analysis.

But I guess current code, most of the time there is no withdraw request at all. And then creating on would definitively be more expensive.

Reply:
>But I guess current code, most of the time there is no withdraw request at all.

If there is no current withdraw request, 

`balanceStates[user][token].pendingWithdraws.stateIndex = 0 < currentStateIndex`

and therefore you are setting it to currentStateIndex anyway (what I think you mean by "creating one")

Reply:
Good call. I should change that... either make the check whether it is set to zero or taking your advice and always set it. I will create an issue.